src/sketchup-stl/exporter.rb

```diff
@@ -130,14 +141,17 @@ def self.write_face_binary(file, scale, mesh, normal)
         polygons = mesh.polygons
         polygons.each do |polygon|
           if (polygon.length == 3)
-            norm = mesh.normal_at(polygon[0].abs)
-            file.write(norm.to_a.pack("e3"))
+            # e - Float: single-precision, little endian byte order
+            file.write(normal.to_a.pack("e3"))
             for j in vertex_order do
               pt = mesh.point_at(polygon[j].abs)
               pt = pt.to_a.map{|e| e * scale}
               file.write(pt.pack("e3"))
             end
-            file.write([0].pack("v"))
+            # 2-byte "Attribute byte count" spacer. Nonstandard use by some stl software
+            # to store color data. Was never widely supported. Should be 0. 
+            # C - Integer: 8-bit unsigned
+            file.write([0, 0].pack("CC"))
```

Comment:
The STL specs says that this is an unsigned 16bit int. It's not two separate bytes.

http://www.fabbers.com/tech/STL_Format#Sct_binary
https://en.wikipedia.org/wiki/STL_(file_format)#Binary_STL

The way I read the specs is that this uint16_t is the size of an optional data section that can follow the facet data. Though some seem to have mis-used it as two "free" bytes.

For this reason I think this should be reversed back to `[0].pack('v')` as it correctly reflect the file specs.